### PR TITLE
Delete tarball after image builds

### LINKED ISSUE #488: We don't delete temporary build tarballs
I have huge residual tarball files in my `/tmp` from my previous `kedge build` runs. I think we should clean these before exiting -

```console
du -sh /tmp/kedge-image-build-*
224M    /tmp/kedge-image-build-295106177
224M    /tmp/kedge-image-build-334355857
224M    /tmp/kedge-image-build-837812191
224M    /tmp/kedge-image-build-878419402
224M    /tmp/kedge-image-build-885045163
```

### PR DESCRIPTION
**Pull request by surajnarwade**
Resolves #488, This PR will delete temporary files created in `/tmp`
for building images with `kedge build`

### CODE REVIEW

File: pkg/build/build.go [hunk 1]
```diff
@@ -101,12 +101,13 @@ func (c *Build) BuildImage(dockerfile, image, context string) error {
...
 	// Build it!
 	if err := c.Client.BuildImage(opts); err != nil {
+		log.Debugf("Deleted temporary file %v", os.Remove(tmpFile.Name()))
 		return errors.Wrap(err, "unable to build image")
 	}
 
 	log.Infof("Image '%s' from directory '%s' built successfully", image, path.Base(context))
 	log.Debugf("Image %s build output:\n%s", image, outputBuffer)
-
+	log.Debugf("Deleted temporary file %v", os.Remove(tmpFile.Name()))
```
Comment (kadel) [inline pkg/build/build.go]: This will show `Deleted temporary file <nil>`

`os.Remove` returns error. You should check it, not printing it.

Comment (kadel) [reply pkg/build/build.go]: This is also nice case for using `defer` to avoid having to call in two places in case of erro

**Comment by surajnarwade** [comment]
@kadel , got it, updated the code :)

File: pkg/build/build.go [hunk 2]
```diff
@@ -99,14 +99,15 @@ func (c *Build) BuildImage(dockerfile, image, context string) error {
 		Dockerfile:   dockerfile,
 	}
 
+	// Delete tarball after creating image
+	defer os.Remove(tmpFile.Name())
```
Comment (kadel) [inline pkg/build/build.go]: defer should be placed right after `tmpFile` is created. Otherwise, it won't be run before for any return between this and  	`tmpFile, err := ioutil.TempFile("/tmp", "kedge-image-build-")`

File: pkg/build/build.go [hunk 3]
```diff
@@ -99,14 +99,15 @@ func (c *Build) BuildImage(dockerfile, image, context string) error {
...
+
 	// Build it!
 	if err := c.Client.BuildImage(opts); err != nil {
 		return errors.Wrap(err, "unable to build image")
 	}
-
```
Comment (surajssd) [inline pkg/build/build.go]: @surajnarwade : unrelated changes!

**Comment by surajnarwade** [comment]
@kadel @surajssd, updated PR with suggested changes, please review
